tests/components/netatmo/test_init.py

```diff
@@ -0,0 +1,224 @@
+        assert await async_setup_component(
+            hass, "netatmo", {"netatmo": {"client_id": "123", "client_secret": "abc"}}
+        )
+
+    await hass.async_block_till_done()
+
+    webhook_id = config_entry.data[CONF_WEBHOOK_ID]
+    await simulate_webhook(hass, webhook_id, FAKE_WEBHOOK_ACTIVATION)
+
+    # Assert webhook is established successfully
+    climate_entity_livingroom = "climate.netatmo_livingroom"
+    assert hass.states.get(climate_entity_livingroom).state == "auto"
+    await simulate_webhook(hass, webhook_id, FAKE_WEBHOOK)
+    assert hass.states.get(climate_entity_livingroom).state == "auto"
+
+
+async def test_setup_with_cloud(hass, config_entry):
+    """Test if set up with active cloud subscription."""
+    await mock_cloud(hass)
+    await hass.async_block_till_done()
+
+    with patch(
+        "homeassistant.components.cloud.async_is_logged_in", return_value=True
+    ), patch(
+        "homeassistant.components.cloud.async_active_subscription", return_value=True
+    ), patch(
+        "homeassistant.components.cloud.async_create_cloudhook",
+        return_value="https://hooks.nabu.casa/ABCD",
+    ) as fake_create_cloudhook, patch(
+        "homeassistant.components.cloud.async_delete_cloudhook"
+    ) as fake_delete_cloudhook, patch(
+        "homeassistant.components.netatmo.api.ConfigEntryNetatmoAuth"
+    ) as mock_auth, patch(
+        "homeassistant.components.netatmo.PLATFORMS", []
+    ), patch(
+        "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation",
+    ), patch(
+        "homeassistant.components.webhook.async_generate_url"
+    ):
+        mock_auth.return_value.post_request.side_effect = fake_post_request
+        assert await async_setup_component(
+            hass, "netatmo", {"netatmo": {"client_id": "123", "client_secret": "abc"}}
+        )
+        assert hass.components.cloud.async_active_subscription() is True
+        fake_create_cloudhook.assert_called_once()
+
+        assert (
+            hass.config_entries.async_entries("netatmo")[0].data["cloudhook_url"]
+            == "https://hooks.nabu.casa/ABCD"
+        )
+
+        await hass.async_block_till_done()
+        assert hass.config_entries.async_entries(DOMAIN)
+
+        for config_entry in hass.config_entries.async_entries("netatmo"):
+            await hass.config_entries.async_remove(config_entry.entry_id)
+            fake_delete_cloudhook.assert_called_once()
+
+        await hass.async_block_till_done()
+        assert len(hass.config_entries.async_entries(DOMAIN)) == 0
+
+
+async def test_api(hass, config_entry):
+    """Test auth instantiation."""
+    config_entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation",
+    ), patch("homeassistant.components.webhook.async_generate_url"):
+        assert await async_setup_component(hass, "netatmo", {})
```

Comment:
We can remove this whole test, right?
```suggestion
```

Reply:
Now it can. I incorporated the coverage of parts of `api.py` in another test.